.github/workflows/build-release.yml

```diff
@@ -0,0 +1,92 @@
+# NOTE: multiple `make` commands could be replaced with
+# `make release` in future if all binaries are built on MacOS machine
+
+name: Build Release binaries
+
+on:
+  # Workflow executes when a new release is created
+  release:
+    types: [created]
+
+jobs:
+  # most binaries can be built on linux machine which is the most cost-efficient on GitHub actions
+  # for building darwin-arm64 binary we need Xcode, therefore, we need to build it on MacOS
+
+  linux-build:
+    name: Build binaries on Ubuntu
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Run Makefile
+        run: |
+          make package
+          make darwin
+          make linux
+          make linux-arm64
+          make windows
+          make windows32
+
+      # Artifacts docs: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts
+      - name: Upload files as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ubuntu-build-files
+          path: dist/aws-sso*
+
+  macos-build:
+    name: Build binaries on MacOS
+    runs-on: macos-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Run Makefile
+        run: |
+          make darwin-arm64
+
+      - name: Upload files as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: macos-build-files
+          path: dist/aws-sso*
+
+  sign-and-upload-files:
+    name: Sign and upload binary files
+    runs-on: ubuntu-latest
+
+    # Wait for binary files to be built
+    needs: [linux-build, macos-build]
+
+    steps:
+      - name: Download ubuntu binaries
+        uses: actions/download-artifact@v3
+        with:
+          name: ubuntu-build-files
+          path: dist/
+
+      - name: Download macos binaries
+        uses: actions/download-artifact@v3
+        with:
+          name: macos-build-files
+          path: dist/
+
+      # Source: https://github.com/crazy-max/ghaction-import-gpg
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v5
```

Comment:
oh, GitHub doesn't have an official action for this?  Not a huge fan of this to be honest.  Minimally it needs to be a specific tag so I don't run arbitrary code with my key/passphrase.

Reply:
Makes sense, will update this as well. Sadly, GitHub does not have any official workflow for handling GPG keys...